### PR TITLE
Signup: Remove the new-launch flow

### LINKED ISSUE #54409: Launch: remove the 'new-launch' flow
### What
The `/start/new-launch` flow was introduced to align with the aesthetics of Gutenboarding signup flow. Now, with re-skin in place, the default `/start/launch-site` is already doing that.

This will also fix `new-launch` related bugs such as https://github.com/Automattic/wp-calypso/issues/42510

List of PRs that introduced Frankenflow and later `new-launch` flow:
- https://github.com/Automattic/wp-calypso/pull/39045
- https://github.com/Automattic/wp-calypso/pull/39434
- https://github.com/Automattic/wp-calypso/pull/39447
- https://github.com/Automattic/wp-calypso/pull/40813
- https://github.com/Automattic/wp-calypso/pull/40860
- https://github.com/Automattic/wp-calypso/pull/41205
- https://github.com/Automattic/wp-calypso/pull/41437 

### Why
tl;dr: One flow is better than two 😄 
Provide a more consistent launch experience and reduce the maintenance overhead.

### A/C
- [ ] Use the default `launch-site` flow for all sites, including the ones created in `/new` (where `site-creation-flow` site option is set to `gutenboarding`). Hint: cleanup [launchSiteOrRedirectToLaunchSignupFlow](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/sites/launch/actions.js#L43)
- [ ] Remove the ['new-launch' flow ](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/config/flows-pure.js#L361-L369) from signup config
- [ ] Cleanup all the customization that was needed for the `new-launch` flow.
  * `isPreLaunch`
  * `isGutenboardingCreate`
  * `getGutenboardingHeader`
  * `isNewLaunchFlow`
  * `isCheckoutFromGutenboarding`
  * ...

### PR DESCRIPTION
**Pull request by chriskmnds**
#### Changes proposed in this Pull Request

Addresses https://github.com/Automattic/wp-calypso/issues/54409 
Removes traces of the "new-launch" flow, along with relevant/specific dependencies and other flags.

#### Testing instructions

TBD

- [x] Confirm that `/start/new-launch` is no longer functional
- [x] Confirm that other flows have not been affected e.g. complete a site launch via `/start`

Fixes https://github.com/Automattic/wp-calypso/issues/54409

### CODE REVIEW

**Review by worldomonation — APPROVED** [review]
Looks okay from the Quality Squad's perspective - the change is limited to a comment, so it should not affect the test.

File: packages/launch/src/focused-launch/domain-details/index.tsx [hunk 1]
```diff
@@ -36,7 +35,7 @@ const DomainDetails: React.FunctionComponent = () => {
 
 	const launchFlowUrl = getCurrentLaunchFlowUrl();
 	const redirectToUseDomainFlow = (): void => {
-		const useYourDomainUrl = `/start/new-launch/domains-launch/use-your-domain?siteSlug=${ siteSubdomain?.domain }&source=${ launchFlowUrl }`;
+		const useYourDomainUrl = `/start/launch-site/domains-launch/use-your-domain?siteSlug=${ siteSubdomain?.domain }&source=${ launchFlowUrl }`;
```
Comment (chriskmnds) [inline packages/launch/src/focused-launch/domain-details/index.tsx:38]: Note to self: Confirm if redirecting to `/start/launch-site/...` is correct and whether `source` is still needed in the query string (not defined in `launch-site` deps).

File: client/my-sites/marketplace/pages/marketplace-domain-upsell/index.tsx [hunk 2]
```diff
@@ -97,7 +97,7 @@ function CalypsoWrappedMarketplaceDomainUpsell(): JSX.Element {
 
 	const redirectToUseDomainFlow = (): void => {
 		const currentUrl = '/plugins/domain';
-		const useYourDomainUrl = `/start/new-launch/domains-launch/use-your-domain?siteSlug=${ selectedSite?.slug }&source=${ currentUrl }`;
+		const useYourDomainUrl = `/start/launch-site/domains-launch/use-your-domain?siteSlug=${ selectedSite?.slug }&source=${ currentUrl }`;
```
Comment (chriskmnds) [inline client/my-sites/marketplace/pages/marketplace-domain-upsell/index.tsx:100]: Note to self: Confirm if redirecting to `/start/launch-site/...` is correct and whether `source` is still needed in the query string (not defined in `launch-site` deps).

File: packages/launch/src/focused-launch/domain-details/index.tsx (hunk 1 above)
Comment (razvanpapadopol) [reply packages/launch/src/focused-launch/domain-details/index.tsx:38]: `source` in this case is used to add a link to the Back button of `use-your-domain` screen and it has nothing to do with the `source` param being required in `providesDependenciesInQuery`

See more on how `source` is used in `client/signup/steps/domains/utils.js`

Screenshot 2021-07-22 at 15 58 35

File: client/state/sites/launch/actions.js [hunk 3]
```diff
@@ -35,13 +35,5 @@ export const launchSiteOrRedirectToLaunchSignupFlow = ( siteId ) => ( dispatch,
...
-
-	const isGutenboarding = [ 'gutenboarding', 'gutenboarding-site-editor' ].includes(
-		getSiteOption( getState(), siteId, 'site_creation_flow' )
-	);
-	if ( isGutenboarding ) {
-		window.location.href = `/start/new-launch?siteSlug=${ siteSlug }&source=home`;
-	} else {
-		window.location.href = `/start/launch-site?siteSlug=${ siteSlug }`;
-	}
+	window.location.href = `/start/launch-site?siteSlug=${ siteSlug }`;
```
Comment (razvanpapadopol) [inline client/state/sites/launch/actions.js:38]: Note: after this change, all the new and existing unlaunched sites created with `/new` will get the `launch-site` flow when launching.

File: client/signup/config/steps-pure.js [hunk 4]
```diff
@@ -654,7 +654,7 @@ export function generateSteps( {
 			stepName: 'launch',
 			apiRequestFunction: launchSiteApi,
 			dependencies: [ 'siteSlug' ],
-			providesDependencies: [ 'isPreLaunch' ],
+			providesDependencies: [],
```
Comment (razvanpapadopol) [inline client/signup/config/steps-pure.js]: I think that `providesDependencies` prop can be removed as well, at least by looking at the config of other steps.

**Review by razvanpapadopol — APPROVED** [review]
LGTM. Nice cleanup ✨

File: client/signup/config/steps-pure.js (hunk 4 above)
Comment (chriskmnds) [reply client/signup/config/steps-pure.js]: Good - guess I went by the opposite observation here without further ado :D
